code/game/objects/structures/votingbox.dm

```diff
@@ -171,38 +172,41 @@
 		else
 			results[text] += 1
 	sortTim(results, cmp=/proc/cmp_numeric_dsc, associative = TRUE)
-
-	var/obj/item/paper/P = new(drop_location())
-	var/list/tally = list()
-	tally += {"
-		<style>
-			.vote_box_content{
-				max-width:250px;
-				display:inline-block;
-				overflow:hidden;
-				text-overflow:ellipsis;
-				white-space:nowrap;
-				vertical-align:bottom
-			}
-			.vote_box_content br {
-				display: none;
-			}
-			.vote_box_content hr {
-				display: none;
-			}
-		</style>
-		"}
-
-	tally += "<h1>Voting Results:</h1><hr><ol>"
-	for(var/option in results)
-		tally += "<li>\"<div class='vote_box_content'>[option]</div>\" - [results[option]] Vote[results[option] > 1 ? "s" : ""].</li>"
-	tally += "</ol>"
-
-	P.info = tally.Join()
-	P.name = "Voting Results"
-	P.update_icon()
-	user.put_in_hands(P)
-	to_chat(user,"<span class='notice'>[src] prints out the voting tally.</span>")
+	if(cooldown > world.time || !can_interact(user))
+		return
+	else
```

Comment:
This whole section doesn't need to be in an else block because the `return` will return from the proc anyway.

Why can't the `can_interact` check live inside `/obj/structure/votebox/Topic`? Currently all of the other hrefs are usable at any range.

Reply:
It certainly can live up there. I'll move it.